lib/RTCSession/ReferSubscriber.js

```diff
@@ -52,6 +52,11 @@ module.exports = class ReferSubscriber extends EventEmitter
 
     extraHeaders.push(referTo);
 
+    // Referred-By header field.
+    const referredBy = `Referred-By: <${this._session._local_identity._uri._scheme}:${this._session._local_identity._uri._user}@${this._session._local_identity._uri._host}>`;
```

Comment:
You don't have to build the URI yourself. Take it from session's contact URI

```js
this._session._ua.contact.uri
```

Reply:
AFAIU it must be the From URI rather than the Contact URI.

Reply:
https://tools.ietf.org/html/rfc3892?#section-3

"It carries a SIP URI representing the  identity of the referrer"

Doesn't say it must be taken from From URI.

Reply:
The identity is in the From. The Contact is just the device location.

Reply:
I pushed a change which crafts the uri from the configuration uri. Please let me know if you want me to make further changes. Thanks!